v-next/hardhat-node-test-reporter/src/error-formatting.ts

```diff
@@ -96,10 +97,27 @@ function formatSingleError(
 
   const diff = getErrorDiff(error);
 
-  const stackReferences: StackReference[] = stackLines
+  let stackReferences: StackReference[] = stackLines
     .map(parseStackLine)
     .filter((reference) => reference !== undefined);
 
+  // Remove all the stack references exposing test runner internals
+  // User code starts after: (Suite|Test|TestHook|...).runInAsyncScope
+  const runInAsyncScopeIndex = stackReferences.findIndex((reference) => {
+    return (
+      reference.context !== undefined &&
+      reference.context.endsWith(".runInAsyncScope")
+    );
+  });
+  if (runInAsyncScopeIndex !== -1) {
```

Comment:
I think we should also check if it's the last entry, and not remove it in that case. Not sure if that would ever happen, but if it does, it would be confusing, as we would show no stack trace.

Reply:
I'll remove that check altogether. But I'll apply the sentiment to how we handle the node references i.e. if references from node is all we have, we should display them to the user.